htdocs/map-google.js

```diff
@@ -0,0 +1,287 @@
+    var iw = getInfoWindow()
+    iw.setContent(
+        '<h3>' + marker.config['receiver_name'] + '</h3>' +
+        '<div>Receiver Location</div>'
+    );
+    iw.open(map, marker);
+};
+
+//
+// GOOGLE-SPECIFIC MAP MANAGER METHODS
+//
+
+MapManager.prototype.setReceiverName = function(name) {
+    if (receiverMarker) receiverMarker.setOptions({ title: name });
+}
+
+MapManager.prototype.removeReceiver = function() {
+    if (receiverMarker) receiverMarker.setMap();
+}
+
+MapManager.prototype.initializeMap = function(receiver_gps, api_key, weather_key) {
+    var receiverPos = { lat: receiver_gps.lat, lng: receiver_gps.lon };
+
+    if (map) {
+        receiverMarker.setOptions({
+            map      : map,
+            position : receiverPos,
+            config   : this.config
+        });
+    } else {
+        var self = this;
+
+        // After Google Maps API loads...
+        $.getScript("https://maps.googleapis.com/maps/api/js?key=" + api_key).done(function() {
+            // Create a map instance
+            map = new google.maps.Map($('.openwebrx-map')[0], {
+                center : receiverPos,
+                zoom   : 5,
+            });
+
+            // Load and initialize day-and-night overlay
+            $.getScript("static/lib/nite-overlay.js").done(function() {
+                nite.init(map);
+                setInterval(function() { nite.refresh() }, 10000); // every 10s
+            });
+
+            // Load and initialize OWRX-specific map item managers
+            $.getScript('static/lib/GoogleMaps.js').done(function() {
+                // Process any accumulated updates
+                self.processUpdates(updateQueue);
+                updateQueue = [];
+            });
+
+            // Create map legend selectors
+            var $legend = $(".openwebrx-map-legend");
+            self.setupLegendFilters($legend);
+            map.controls[google.maps.ControlPosition.LEFT_BOTTOM].push($legend[0]);
+
+            // Create receiver marker
+            if (!receiverMarker) {
+                receiverMarker = new google.maps.Marker();
+                receiverMarker.addListener('click', function() {
+                    showReceiverInfoWindow(receiverMarker);
+                });
+            }
+
+            // Set receiver marker position, name, etc.
+            receiverMarker.setOptions({
+                map      : map,
+                position : receiverPos,
+                title    : self.config['receiver_name'],
+                config   : self.config
+            });
+        });
+    }
+};
+
+MapManager.prototype.processUpdates = function(updates) {
+    var self = this;
+
+    if (typeof(GMarker) === 'undefined') {
+        updateQueue = updateQueue.concat(updates);
+        return;
+    }
+
+    updates.forEach(function(update) {
+        if (typeof update.source === 'undefined' || typeof update.source.callsign === 'undefined') {
+            console.error(update);
```

Comment:
this will need some updates for ADS-B. currently all ADS-B updates land here and don't show on the map.

Reply:
Update: this seems to apply to all types of plane sources (i.e. ADS-B, HFDL, VDL2). We can work on some better distinction between the different types of sources instead of just switching on available properties.

Reply:
in latest commit I've started the work for the Plane markers, but I do not have the knowledge to finish it right now.
I would appreciate some help here or I can investigate further and eventually finish it at some time later.
[start here](https://github.com/jketterl/openwebrx/pull/354/commits/970f7476e3cf5c8495f52c0ad97eb6b57137ef11#diff-5a856e46e8e874f4ef7bd7130214d2193c751aaba7f3cd3461b471bac594d443R192)